packages/auth/src/authClients/supertokens.ts

```diff
@@ -0,0 +1,42 @@
+import type { AuthClient } from './'
+
+export interface SuperTokensUser {
+  userId: string
+  accessTokenPayload: any
+}
+
+export type SuperTokens = AuthClient
+
+export const supertokens = (client: { authRecipe: any, sessionRecipe: any }): AuthClient => {
```

Comment:
It would be helpful to the user to not use `any`

Reply:
So there are various `authRecipe` that we have and the user could be using any of them. They are essentially classes with various static functions. Since we don't know which authRecipe a user may end up using, we have put them as `any`.

Reply:
Hmmmm 🤔 maybe there's a base recipe or base sessionRecipe interface? 

Given the implementation here, I think the expected type for authRecipe is an object that _atleast_ contains `redirectToAuth(action: string)`, `signOut` for example.

Reply:
@dac09 Updated the typing

Reply:
Yea. There isn't a base recipe. We will have to change our SDK to add such an interface to it. Furthermore, since they are static functions, an interface wouldn't solve any purpose anyway. 

What we could do is is something like:
```ts
import type EmailPassword from "supertokens-auth-react/recipe/emailpassword;
import type ThirdParty from "supertokens-auth-react/recipe/thirdparty;
import type ThirdPartyEmailPassword from "supertokens-auth-react/recipe/thirdpartyemailpassword;
import type Session from "supertokens-auth-react/recipe/session;
```

And the in the types, we could do:

```ts
authRecipe:  EmailPassword | ThirdParty | ThirdPartyEmailPassword, sessionRecipe: Session`
```

But then everytime we add a new recipe, we will have to create a PR for redwood. What do you suggest?

Reply:
Nvm. Ignore my comment, I think we have created types for authRecipe and sessionRecipe.